Update comment on deprecation warning to point inernal users to use safer API

kimishpatel opened the pull request:
Summary:
For Meta internal use there is UseNThreadsThreadPoolGuard API to allow users to use a subset of threads. Point users to this API.

Created from CodeHub with https://fburl.com/edit-in-codehub

Reviewed By: swolchok

Differential Revision: D77973302

swolchok commented on extension/threadpool/threadpool.h:
```diff
@@ -42,8 +42,11 @@ class ThreadPool final {
+   * to run the model on.
    */
-  [[deprecated("This API is experimental and may change without notice.")]]
+  [[deprecated("This API is experimental and may change without notice. Consider using UseNThreadsThreadPoolGuard")]]
```
nit: period at end of sentence.

kimishpatel replied:
good point. forgot about this.

kimishpatel committed:
Update comment on deprecation warning to point inernal users to use safer API (#12318)

Summary:

For Meta internal use there is UseNThreadsThreadPoolGuard API to allow users to use a subset of threads. Point users to this API.

Created from CodeHub with https://fburl.com/edit-in-codehub

Reviewed By: swolchok

Differential Revision: D77973302
